### PR TITLE
add ocaml opium

### PR DESCRIPTION
**Pull request by mudrz**
follow up on https://github.com/TechEmpower/FrameworkBenchmarks/pull/6057

that builds on top of https://github.com/TechEmpower/FrameworkBenchmarks/pull/6041 and https://github.com/TechEmpower/FrameworkBenchmarks/pull/6060

this PR adds benchmarks for opium - an OCaml web server:
- plaintext
- json
- queries
- updates
- fortunes

### CODE REVIEW

**Comment by rbjorklin** [comment]
Hey @mudrz! I'm not familiar with opium but don't you need a main function as an entrypoint to your program like I have for [webmachine](https://github.com/TechEmpower/FrameworkBenchmarks/blob/master/frameworks/OCaml/webmachine/src/tfb.ml#L185-L188)? On top of that I'm pretty sure you need a [dune](https://github.com/TechEmpower/FrameworkBenchmarks/blob/master/frameworks/OCaml/webmachine/src/dune#L1) file that says `executable` somewhere in the PR.

**Comment by mudrz** [comment]
omg, thanks @rbjorklin I wasted so much time on this and didn't even notice this...
so much wasted time.........

in the main repo `.gitignore` there is an ignore for `bin` "common junk"
https://github.com/TechEmpower/FrameworkBenchmarks/blob/69b5ba56fe3bc1495b5d27ee3f9452fed5cec6f2/.gitignore#L25-L32

which unfortunately is the common folder for putting dune executable, so what has been happening is that the main code of my project was not committed to the branch since it is `bin/main.ml` 🤦‍♂️

**Comment by mudrz** [comment]
added the missing dir...
it also has a date implementation that you can use to avoid the warning
```ocaml
module Time = struct
  let weekday Unix.{tm_wday;_} = match tm_wday with
    | 0 -> "Sun"
    | 1 -> "Mon"
    | 2 -> "Tue"
    | 3 -> "Wen"
    | 4 -> "Thu"
    | 5 -> "Fri"
    | 6 -> "Sat"
    | _ -> failwith "weekday"

  let month Unix.{tm_mon;_} = match tm_mon with
    | 0 -> "Jan"
    | 1 -> "Feb"
    | 2 -> "Mar"
    | 3 -> "Apr"
    | 4 -> "May"
    | 5 -> "Jun"
    | 6 -> "Jul"
    | 7 -> "Aug"
    | 8 -> "Sep"
    | 9 -> "Oct"
    | 10 -> "Nov"
    | 11 -> "Dec"
    | _ -> failwith "month"

  let gmt tm =
    Printf.sprintf "%s, %02u %s %04u %02u:%02u:%02u GMT" (weekday tm) tm.tm_mday (month tm) (tm.tm_year + 1900)  tm.tm_hour tm.tm_min tm.tm_sec

  let now () = (gmt (Unix.gmtime (Unix.gettimeofday ())))
end
```

**Comment by mudrz** [comment]
@nbrady-techempower PTAL, this adds a new OCaml web server

**Comment by mudrz** [comment]
@rbjorklin fyi added fortunes and a bad updates implementation (without batching)

**Comment by rbjorklin** [comment]
Hey @mudrz! I just found this: https://github.com/TechEmpower/FrameworkBenchmarks/issues/6064 which means the next round of tests will be interrupted and there might be some disruption to the test suite over the coming days/weeks.

**Comment by mudrz** [comment]
oh, okay, thanks for letting me know, I was hoping to see how a standard OCaml server performs under a standardised test, but it's good that the tool will also improve

this can still be merged though and we'll see the comparison in the near future

File: frameworks/OCaml/opium/src/exe/tfb_headers.ml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+module Time = struct
+  let weekday Unix.{tm_wday;_} = match tm_wday with
+    | 0 -> "Sun"
+    | 1 -> "Mon"
+    | 2 -> "Tue"
+    | 3 -> "Wen"
```
Comment (ryanslade) [inline frameworks/OCaml/opium/src/exe/tfb_headers.ml:6]:
```suggestion
    | 3 -> "Wed"
```

Comment (mudrz) [reply frameworks/OCaml/opium/src/exe/tfb_headers.ml]: thanks @ryanslade

**Pull request merged**